dali/test/python/operator_2/test_random_bbox_crop.py

```diff
@@ -494,6 +510,64 @@ def get_boxes():
         pipe.run()
 
 
+def test_crop_window_warning():
+    n_boxes = [10, 12, 0]
+    pipe = Pipeline(batch_size=len(n_boxes), num_threads=1, device_id=0)
+
+    def get_boxes():
+        return [np.random.uniform(0, 1, size=(n, 4)).astype(np.float32) for n in n_boxes]
+
+    boxes = fn.external_source(source=get_boxes)
+    n_attempt = 10
+    processed = fn.random_bbox_crop(
+        boxes,
+        thresholds=[1.0],  # Impossible threshold to force the warning
+        bbox_layout="xyXY",
+        allow_no_crop=False,
+        total_num_attempts=n_attempt,
+    )
+    pipe.set_outputs(*processed)
+    with TemporaryFile(mode="w+") as f:
+        with redirect_stderr_fd(f.fileno()):
```

Comment:
Have you considered using `check_output_pattern` from `test_utils`?

Reply:
That is a good point, I didn't check for any existing internal tools. However, I am looking to validate that the string appears exactly twice, or in the later case doesnt appear, `check_output_pattern` doesn't quite match this use case.
I suppose a regex could be built to match this, but this seems contrived compared to len(lines) == 2 and all(l == expected for l in lines).

Reply:
`^$` should match empty input. If you feel that it deserves a dedicated `redirect_stderr_fd`, I would include `TemporaryFile` inside, and maybe pass a validator function that accepts file handle as an argument and runs various checks on its content. In effect you wouldn't miss adding `redirect_stderr_fd` in the second test.

Reply:
> ^$ should match empty input. If you feel that it deserves a dedicated redirect_stderr_fd, I would include TemporaryFile inside, and maybe pass a validator function that accepts file handle as an argument and runs various checks on its content. In effect you wouldn't miss adding redirect_stderr_fd in the second test.

I really walked into that 😆